tests/unit/test_coordinate_utilities.py

```diff
@@ -0,0 +1,431 @@
+
+    def test_get_override_projected_dimensions(self):
+        """Ensure that the expected projected dimension name
+        is returned for the coordinate variables
+        """
+
+        expected_override_dimensions_AM = [
+            '/Soil_Moisture_Retrieval_Data_AM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_AM/projected_x',
+        ]
+        expected_override_dimensions_PM = [
+            '/Soil_Moisture_Retrieval_Data_PM/projected_y',
+            '/Soil_Moisture_Retrieval_Data_PM/projected_x',
+        ]
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the science variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_AM/surface_flag'
+                ),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the longitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.longitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions for the latitude variable'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(self.varinfo, self.latitude),
+                expected_override_dimensions_AM,
+            )
+
+        with self.subTest(
+            'Retrieves expected override dimensions science variable with a different grid'
+        ):
+            self.assertEqual(
+                get_override_projected_dimensions(
+                    self.varinfo, '/Soil_Moisture_Retrieval_Data_PM/surface_flag_pm'
+                ),
+                expected_override_dimensions_PM,
+            )
+
+    def test_get_row_col_sizes_from_coordinate_datasets(self):
+        """Ensure that the correct row and column sizes are
+        returned for the requested coordinates
+        """
+        expected_row_col_sizes = (5, 10)
+        lat_1d_array = np.array([1, 2, 3, 4])
+        lon_1d_array = np.array([6, 7, 8, 9])
+        lat_irregular_array = np.array([[1, 2, 3], [3, 4, 5]])
+        lon_irregular_array = np.array([[6, 7], [8, 9], [10, 11]])
+        with self.subTest('Retrieves the expected row col sizes from the coordinates'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(self.lat_arr, self.lon_arr),
+                expected_row_col_sizes,
+            )
+        with self.subTest('Retrieves the expected row col sizes for the 1d array'):
+            self.assertEqual(
+                get_row_col_sizes_from_coordinate_datasets(lat_1d_array, lon_1d_array),
+                (4, 4),
+            )
+
+        with self.subTest(
+            'Raises an exception when the lat and lon array shapes do not match'
+        ):
```

Comment:
I think there are a few other subtests for when things aren't quite as expected/needed:

* Both arrays are 1-D, but latitude is has a zero size. (Probably using your `np.empty((0, ))` trick.) Should raise an exception.
* Both arrays are 1-D, but latitude is has a zero size. Should raise an exception.
* Somehow having a latitude array that is zero dimensional (`np.empty(())`). Should raise an exception.
* Somehow having a longitude array that is zero dimensional. Should raise an exception.

It seems like a fair bit of duplication, but you specify separate logical checks against latitude and longitude, and you have a lot of different checks overall in that function. The cases above are isolating different parts of those AND/OR combined conditionals. Hopefully, with something like `np.empty(...)` it's quick to cobble together some extra tests, and you'll be able to re-use the `self.assertRaises` scaffolding you already have in the subtest below.

Reply:
updated - 36a3c4034215d85f5fccf660dc3f213d28873d3d